crates/aranya-fast-channels/src/shm/shared.rs

```diff
@@ -830,7 +834,9 @@ impl<CS: CipherSuite> ChanListData<CS> {
     }
 
     fn len(&self) -> Result<usize, Corrupted> {
-        usize::try_from(self.len).map_err(|_| corrupted("`len` is larger than `usize::MAX`"))
+        let len = isize::try_from(self.len)
+            .map_err(|_| corrupted("`len` is larger than `isize::MAX`"))?;
+        usize::try_from(len).map_err(|_| corrupted("`len` conversion to usize failed"))
```

Comment:
this actually might even work as `isize::try_from(self.len).map_err(...).try_into().map_err(...)`